src/Compilers/CSharp/Test/Semantic/Semantics/LambdaTests.cs

```diff
@@ -2474,6 +2474,61 @@ void Main()
                 );
         }
 
+        [Fact, WorkItem(22662, "https://github.com/dotnet/roslyn/issues/22662")]
+        public void LambdaSquigglesArea()
+        {
+            var src = @"
+class C
+{
+    void M()
+    {
+        System.Func<bool, System.Action<bool>> x = x1 => x2 =>
+        {
+            error();
+        };
+    }
+}
+";
+            var comp = CreateStandardCompilation(src);
+            comp.VerifyDiagnostics(
+                // (8,13): error CS0103: The name 'error' does not exist in the current context
+                //             error();
+                Diagnostic(ErrorCode.ERR_NameNotInContext, "error").WithArguments("error").WithLocation(8, 13),
+                // (6,58): error CS1662: Cannot convert lambda expression to intended delegate type because some of the return types in the block are not implicitly convertible to the delegate return type
+                //         System.Func<bool, System.Action<bool>> x = x1 => x2 =>
+                Diagnostic(ErrorCode.ERR_CantConvAnonMethReturns, "x2 =>").WithArguments("lambda expression").WithLocation(6, 58)
+                );
```

Comment:
Does this test catch the change? The location is the start of the span only.

Reply:
Yes, this test behaves differently with the fix. The squiggled region is quoted: "x =>".
Without the fix, this `Diagnostic` is multi-line.

Reply:
The test is checking the start of the span only though.

Reply:
My mistake, the text in the span is one of the arguments to `Diagnostic`.

Reply:
I'll investigate

Reply:
I confirmed that the test framework is sensitive to changes in the `squiggledText` part of the `Diagnostic`.